MainForm.cs

```diff
@@ -0,0 +1,173 @@
+
+                label3.Text = "Выиграла комбинация № " + Service.WinCombination.NumOfWinner;
+                resCard.BackColor = ColorTranslator.FromHtml(Colors.GetDisplayName(Service.WinCombination.WinnerCard.Color));
+                resC.Text = Convert.ToString(Service.WinCombination.WinnerCard.Nominal);
+                if (Service.WinCombination.NumOfWinner == 1)
+                {
+                    textBox1.BackColor = Color.LightGreen;
+                } 
+                else
+                {
+                    textBox2.BackColor = Color.LightGreen;
+                }
+                for (int i = 0; i < combLen1; i++)
+                {
+                    labelsOnCards1[i].Text = Convert.ToString(cards1[i].Nominal);
+                    cardPanels1[i].BackColor = ColorTranslator.FromHtml(Colors.GetDisplayName(cards1[i].Color));
+                }
+                for (int i = 0; i < combLen2; i++)
+                {
+                    labelsOnCards2[i].Text = Convert.ToString(cards2[i].Nominal);
+                    cardPanels2[i].BackColor = ColorTranslator.FromHtml(Colors.GetDisplayName(cards2[i].Color));
+                }
+            }
+
+        }
+
+        private void DoBtnClick(object sender, EventArgs e)
+        {
+            bool isSizeInt = true;
+            errorMessageLabel.Text = "";
+
+            try
+            {
+                bool isCombLen1_Correct = int.TryParse(textBox1.Lines[0], out int combLen1);
+                bool isCombLen2_Correct = int.TryParse(textBox2.Lines[0], out int combLen2);
+
+                if (!isCombLen1_Correct)
+                {
+                    textBox1.ForeColor = Color.Red;
+                    textBox1.Text = "\nInvalid value of count of cards!";
+                    isSizeInt = false;
+                }
+                if (!isCombLen2_Correct)
+                {
+                    textBox2.ForeColor = Color.Red;
+                    textBox2.Text = "\nInvalid value of count of cards!";
+                    isSizeInt = false;
+                }
+                if (isSizeInt)
+                {
+                    bool isSizeCorrect = true;
+                    if (combLen1 > 7 || combLen1 < 1)
+                    {
+                        textBox1.ForeColor = Color.Red;
+                        textBox1.Text = "\nCount of cards should be in the range [1, 7]!";
+                        isSizeCorrect = false;
+                    }
+                    if (combLen2 > 7 || combLen2 < 1)
+                    {
+                        textBox2.ForeColor = Color.Red;
+                        textBox2.Text = "\nCount of cards should be in the range [1, 7]!";
+                        isSizeCorrect = false;
+                    }
+                    if (isSizeCorrect)
+                    {
+                        MainFunc(combLen1, combLen2);
+                    }
+                }
+            }
+            catch(Exception error)
+            {
```

Comment:
Are you sure that all unexpected errors should tell to user `Error: an empty string was found instead of the expected value!` message? It can be some other case. If you expect some specific Exception, better will be to use some specific Exception class. It may be some Custom exception created for this case